content/en/docs/reference/using-api/cel.md

```diff
@@ -0,0 +1,325 @@
+
+A `authorizer.requestResource` variable is also available for some CEL
+expressions. This variable is configured to check permissions of a requested
+resource.
+
+Check the API documentation of fields where CEL is used to learn if
+authorization variables are available for a particular field.
+
+| CEL Expression                                                                         | Purpose                                                                                                               |
+|----------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------|
+| `authorizer.group('').resource('pods').namespace('default').check('create').allowed()` | Check if the principal (user or service account) submitting a request is authorized to create pods in the default namespace.                         |
+| `authorizer.path('/healthz').check('get').allowed()`                                   | Check if the principal (user or service account) submitting the request is authorized to to perform a HTTP GET for the non-resource `/healthz` path. |
+| `authorizer.requestResource.check('custom-verb').allowed()`                            | Checks if the principal (user or service account) submitting the request is authorized for `custom-verb` on the resource being handled.              |
+
+See the [Authorization Library godoc](https://pkg.go.dev/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/library#Authz) for more information.
+
+## Type Checking
+
+CEL is a [gradually typed language](https://github.com/google/cel-spec/blob/master/doc/langdef.md#gradual-type-checking).
+
+Some Kubernetes API fields contain fully type checked CEL expressions. For
+example, [CustomResourceDefinitions Validation
+Rules](/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules)
+are fully type checked.
+
+Some Kubernetes API fields contain partially type checked CEL expressions. A
+partially type checked expression is an experessions where some of the variables
+are statically typed but others are dynamically typed. For example, in the CEL
+expressions of
+[ValidatingAdmissionPolicies](/docs/reference/access-authn-authz/validating-admission-policy/)
+the `request` variable is typed, but the `object` variable is dynamically typed.
+As a result, an expression containing `request.namex` would fail type checking
+because the `namex` field is not defined. However, `object.namex` would pass
+type checking even when the `namex` field is not defined for the resource kinds
+that `object` refers to, because `object` is dynamically typed.
+
+The `has()` macro in CEL may be used in CEL expressions to check if a field of a
+dynamically typed variable is accessable before attempting to access the field's
+value. For example:
+
+```
```

Comment:
You could append `cel` to this line, and then any future syntax highlighting support will pick it up.

Reply:
Oh nice. Thx!